code/obj/artifacts/artifact_objects/bombs.dm

```diff
@@ -34,28 +39,57 @@ ABSTRACT_TYPE(/datum/artifact/bomb)
 
-		var/turf/T = get_turf(O)
+		src.detonation_time = TIME + src.explode_delay
 
+		// this is all just fluff
+		var/turf/T = get_turf(O)
 		if (warning_initial)
 			T.visible_message("<b><span class='alert'>[O] [warning_initial]</b></span>")
 		if (alarm_initial)
-			playsound(T, alarm_initial, 100, 1, -1)
+			playsound(T, alarm_initial, 100, 1, doAlert?200:-1)
+		if (doAlert && !ON_COOLDOWN(O, "alertArm", 10 MINUTES)) // spam protection
+			var/area/A = get_area(O)
+			command_alert("\An extremely unstable object of [artitype.name] origin has been detected in [A]. The crew is advised to dispose of it immediately.", "Station Threat Detected")
+		O.add_simple_light("artbomb", list(255,255,255,255))
+		animate(O, pixel_y = rand(-3,3), pixel_y = rand(-3,3),time = 1,loop = src.explode_delay + 10 SECONDS, easing = ELASTIC_EASING, flags=ANIMATION_PARALLEL)
+		animate(O.simple_light, flags=ANIMATION_PARALLEL, time = src.explode_delay + 10 SECONDS, transform = matrix() * animationScale)
+
+
+	effect_process(var/obj/O)
+		if(src.activated) // repeating noise, so people who come near later know it's a bomb
+			playsound(get_turf(O), alarm_during, 30, 1)
 
-		SPAWN_DBG(src.explode_delay)  //who the fuck coded this shit below without running get_turf again
-			T = get_turf(O)
+		if(TIME > src.detonation_time)
+			src.detonation_time = INFINITY
+			var/turf/T = get_turf(O)
+			if (!O || !T || !src.activated) // please stop
+				return
+
+			// more fluff
 			if (warning_final)
 				T.visible_message("<b><span class='alert'>[O] [warning_final]</b></span>")
 			if (alarm_final)
 				playsound(T, alarm_final, 100, 1, -1)
-			animate_flash_color_fill(O,flascustomization_first_color,10,3)
-
-			sleep(3 SECONDS)
-
-			T = get_turf(O)
-			if (src.activated)
-				deploy_payload(O)
-			else
-				T.visible_message("<b><span class='notice'>[O] [text_disarmed]</b></span>")
+			animate(O, pixel_y = rand(-3,3), pixel_y = rand(-3,3),time = 1,loop = 10 SECONDS, easing = ELASTIC_EASING, flags=ANIMATION_PARALLEL)
+			animate(O.simple_light, flags=ANIMATION_PARALLEL, time = 10 SECONDS, transform = matrix() * animationScale)
+
+			// actual boom
+			spawn(10 SECONDS)
+				if (src.activated)
+					blewUp = 1
+					deploy_payload(O)
+
+	effect_deactivate(obj/O)
+		. = ..()
+		// and remove all the animation stuff when it is deactivated (:
+		animate(O, pixel_y = 0, pixel_y = 0, time = 3,loop = 1, easing = LINEAR_EASING)
+		animate(O.simple_light, flags=ANIMATION_PARALLEL, time= 3 SECONDS, transform = null)
+		spawn(3 SECONDS)
+			O.remove_simple_light("artbomb")
+		var/turf/T = get_turf(O)
+		T.visible_message("<b><span class='notice'>[O] [text_disarmed]</b></span>")
+		if(src.doAlert && !src.blewUp && !ON_COOLDOWN(O, "alertDisarm", 10 MINUTES)) // lol, don't give the message if it was destroyed by exploding itself
+			command_alert("\The object of [src.artitype.name] origin has been neutralized. All personnel should return to their duties.", "Station Threat Neutralized")
```

Comment:
I'd prefer having the command alert only when it's destroyed, else it's a false sense of security (you could activate, deactive, and quickly activate again, right?). Also the `\` isn't necessary.

Reply:
I would really hate not having the message when it gets deactivated, I think I will simply make it so that you can't reactivate it until 10 minutes have passed.

Reply:
That's fair! Thanks for being accommodating (for this and in general)!